modules/core/src/main/scala/org/tessellation/l0/SnapshotService.scala

```diff
@@ -0,0 +1,59 @@
+package org.tessellation.l0
+
+import cats.Applicative
+import cats.effect.kernel.Async
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import org.tessellation.config.types.SnapshotConfig
+import org.tessellation.dag.block.{BlockStorage, BlockValidator}
+import org.tessellation.schema.height.Height
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.PosLong
+import io.estatico.newtype.ops._
+
+trait SnapshotService[F[_]] {
+  def createSnapshot(minTipHeight: PosLong): F[DAGSnapshot]
+}
+
+object SnapshotService {
+
+  def make[F[_]: Applicative: Async](
+    config: SnapshotConfig,
+    blockStorage: BlockStorage[F],
+    blockValidator: BlockValidator[F],
+    snapshotStorage: SnapshotStorage[F]
+  ): SnapshotService[F] =
+    new SnapshotService[F] {
+
+      def createSnapshot(minTipHeight: PosLong): F[DAGSnapshot] =
+        for {
+          previousSnapshotHeight <- snapshotStorage.getLastSnapshotHeight
+          nextSnapshotHeight = Height(previousSnapshotHeight.coerce + config.heightInterval)
+
+          // In 1.0: validateSnapshotHeightIntervalCondition
+          //  if (minTipHeight > nextHeight + config.heightDelayInterval)
+
+          acceptedBlocks <- blockStorage.getAccepted
+
+          blocksForNextSnapshot = acceptedBlocks.filter { block =>
+            val blockHeight = block.signed.height.coerce
+            blockHeight > previousSnapshotHeight.coerce && blockHeight <= nextSnapshotHeight.coerce
+          }
+
+          // In 1.0: validateAcceptedCBsSinceSnapshot
+          //  if (blocksForNextSnapshot.nonEmpty) OK
+
+          // In 1.0: validateBlocksWithinHeightInterval
+          //  blocksForNextSnapshot.toSeq.map(_.signed).traverse(blockValidator.validate)
```

Comment:
Can we assume that accepted blocks are already validated? Why we needed additional validation in 1.0?

Reply:
I would go with single point of validation and strict control over storages at least for now. If possible all blocks should go through the same path to the block storage to have a control over the logic - hence be sure that accepted is known to be valid.

Reply:
keep in mind that some types of blocks (state channels) will be only validated during snapshot creation process

Reply:
While it may seem redundant, we will want every cell to validate input before broadcasting the output. It's possible that invalid/forged blocks can be received from L1 processes. Filtering these out before proposing in a snapshot helps prevent redownloads and will help with convergence and cluster stability in general

Reply:
That's exactly what single point of validation would do I think. The block will not be saved as valid until it is actually valid. Here we are implementing the logic for our DAGBlocks, not some abstract general block, so we can handle them the way we want. For the blocks from other state channels the logic will be for sure different and they can be stored in a different place and they will also have a single point of validation (while being pulled to the snapshot). That's at least how I see it.

Reply:
> While it may seem redundant, we will want every cell to validate input before broadcasting the output. It's possible that invalid/forged blocks can be received from L1 processes. Filtering these out before proposing in a snapshot helps prevent redownloads and will help with convergence and cluster stability in general

I agree with Maciek. Such issue occurs in Global Snapshot and we should filter these blocks out but in my case it is just a DAGSnapshot so if we implement single point of validation in block acceptance (DAG blocks) then we do not need to validate blocks again,